👌 IMPROVE: Rename startRow/maxRows to from/size, respectively.

michaelborn opened the pull request:
New syntax:

```js
searchBuilder.setFrom( 11 );
searchBuilder.setSize( 10 );
```

Still keeps backwards-compatible shims in place for the old parameter names:

```js
var searchBuilder = variables.model.new( variables.testIndexName, "testdocs", {
					maxRows : 15,
					startRow : 16
				} );
searchBuilder.setStartRow( 51 );
searchBuilder.setMaxRows( 50 );
```

- [x] Renames `startRow` to `from` 💪
- [x] Renames `maxRows` to `size` 🔨
- [x] Keeps support for older names in both `.new()` and setter methods 🎉
- [x] Adds test for new names 🤖
- [x] Adds pagination docs to `Search.md` 📖

michaelborn committed:
👌 IMPROVE: Rename startRow/maxRows to from/size, respectively.

Still keeps backwards-compatible shims in place for the old parameter names.

jclausen commented on models/SearchBuilder.cfc:
```diff
@@ -150,6 +150,24 @@ component accessors="true" {
 	}
 
+	/**
+	 * !DEPRECATED - Backwards compatible setter for max result size
```
Use the javadoc `@deprectated` annotation.  Then it shows up better when Docbox generates from the source.

jclausen replied:
Since this is deprecated, can you also replace the usage of `setMaxRows` and `setStartRow` in this repository. I see one `setMaxRows` used in the Migration manager

jclausen commented on models/SearchBuilder.cfc:
```diff
@@ -150,6 +150,24 @@ component accessors="true" {
+		return this;
+	}
+	/**
+	 * !DEPRECATED - Backwards compatible setter for result start offset
```
Same note as above

michaelborn committed:
👌 IMPROVE: Fix @deprecated note for docs, drop usage of deprecated methods outside of tests

michaelborn commented:
Added `@deprecated` note for DocBox, dropped Manager's use of `setMaxRows()`. I see no use of these deprecated methods outside of the tests which are explicit tests for backwards compatibility.

jclausen commented:
@michaelborn This one has merge conflicts now.

michaelborn committed:
Merge branch 'development' into feat/rename-maxrows-startrow
